knife/lib/chef/knife/core/bootstrap_context.rb

```diff
@@ -223,6 +230,15 @@ def first_boot
           end
         end
 
+        def macos_dir
+          Mixlib::Install::Dist::MACOS_VOLUME
+        end
+
+        def download_url
+          format(config[:license_url], config[:channel]) + \
+            "/$project/metadata?v=$version&p=$platform&pv=$platform_version&m=$machine&license_id=#{config[:license_id]}"
```

Comment:
Are `$project` etc. supposed to be interpolations or strings for the shell the process?

Reply:
These are the variables on the shell script to download and install the product. 

https://github.com/chef/chef/pull/14559/files#diff-b477eab4fae6a837d6caa832d015418c2fa1f16287a3cf687f1579c1aeaac748R331